src/mame/drivers/rastersp.cpp

```diff
@@ -709,22 +1223,41 @@ void rastersp_state::io_map(address_map &map)
 	map(0x1010, 0x1013).portr("DSW1");
 	map(0x1014, 0x1017).portr("EXTRA");
 	map(0x4000, 0x4007).rw("rtc", FUNC(mc146818_device::read), FUNC(mc146818_device::write)).umask32(0x000000ff);
+}
+void rastersp_state::io_map(address_map &map)
+{
+	io_map_base(map);
 	map(0x6008, 0x600b).nopr().nopw(); // RS232
 }
 
+void fbcrazy_state::io_map(address_map &map)
+{
+	rastersp_state::io_map_base(map);
+	map(0x6000, 0x600f).rw( FUNC(fbcrazy_state::edc_tb_r), FUNC(fbcrazy_state::edc_tb_w));
+}
 
-void rastersp_state::dsp_map(address_map &map)
+void rastersp_state::dsp_map_base(address_map &map)
 {
-	map(0x000000, 0x0fffff).bankrw("bank1");
 	map(0x400000, 0x40ffff).rom().region("dspboot", 0);
 	map(0x808000, 0x80807f).rw(FUNC(rastersp_state::tms32031_control_r), FUNC(rastersp_state::tms32031_control_w));
 	map(0x880402, 0x880402).w(FUNC(rastersp_state::dsp_unk_w));
 	map(0x883c00, 0x883c00).w(FUNC(rastersp_state::dsp_486_int_w));
 	map(0xc00000, 0xc03fff).bankrw("bank2");
 	map(0xc80000, 0xc80000).w(FUNC(rastersp_state::dsp_ctrl_w));
-	map(0xfc0000, 0xffffff).bankrw("bank3");
+	map(0xfc0000, 0xffffff).bankrw("bank1");//was3
 }
 
+void rastersp_state::dsp_map(address_map &map)
+{
+	dsp_map_base(map);
+	map(0x000000, 0x0fffff).bankrw("bank1");//4 meg
+}
+
+void fbcrazy_state::dsp_map(address_map &map)
+{
+	rastersp_state::dsp_map_base(map);
+	map(0x000000, 0x1fffff).bankrw("bank1");//8 meg
+}
```

Comment:
Is it correct that the DSP and main CPU always see the same bank?

Reply:
I believe so. I can't see anything in the documentation for the  board which suggests they can be different.